File: repositories/license_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type LicenseRepository interface {
	Create(ctx context.Context, license *entities.DoctorLicense) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorLicense, error)
	Delete(ctx context.Context, licenseID, doctorID uuid.UUID) error
}

type GormLicenseRepository struct {
	db *gorm.DB
}

func NewLicenseRepository(db *gorm.DB) LicenseRepository {
	return &GormLicenseRepository{db: db}
}

func (r *GormLicenseRepository) Create(ctx context.Context, license *entities.DoctorLicense) error {
	return r.db.WithContext(ctx).Create(license).Error
}

func (r *GormLicenseRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorLicense, error) {
	var licenses []entities.DoctorLicense
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Find(&licenses).Error
	return licenses, err
}

func (r *GormLicenseRepository) Delete(ctx context.Context, licenseID, doctorID uuid.UUID) error {
	return r.db.WithContext(ctx).Where("license_id = ? AND doctor_id = ?", licenseID, doctorID).Delete(&entities.DoctorLicense{}).Error
}

File: python_backend/repositories/license_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.license import DoctorLicense

class LicenseRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, license: DoctorLicense) -> None:
        self.db.add(license)
        self.db.commit()
        self.db.refresh(license)

    def FindByDoctorID(self, doctor_id: UUID) -> List[DoctorLicense]:
        return self.db.query(DoctorLicense).filter(DoctorLicense.doctor_id == str(doctor_id)).all()

    def Delete(self, license_id: UUID, doctor_id: UUID) -> None:
        self.db.query(DoctorLicense).filter(
            DoctorLicense.license_id == str(license_id),
            DoctorLicense.doctor_id == str(doctor_id),
        ).delete()
        self.db.commit()
